### DISCUSSION TITLE
TOML: Escape text on paste in string literals

### DISCUSSION THREAD

**Discussion by xpe** [intellij-rust/intellij-rust#7677] [IntelliJ Rust Plugin Development]
I'd like to request a feature. I would like the TOML plugin (this repository, correct?) to escape text when pasted in the scope of TOML string literals. I may be able to look into this as a pull request. Please let me know if this is the right repository.

I use CLion 2021.2 on macOS 11.5.1 and see that it has a toggle switch for JSON titled "Escape text on paste in string literals" in Preferences > Editor > General > Smart Keys > JSON.

**Comment by Undin** [comment]
@xpe Hi!

> Please let me know if this is the right repository.

It's correct repo

> I would like the TOML plugin (this repository, correct?) to escape text when pasted in the scope of TOML string literals.

I've created the corresponding issue #7706

> I may be able to look into this as a pull request

It would be great, but I want to mention that we are going to move the development of TOML plugin into https://github.com/JetBrains/intellij-community repo since TOML langauge is used in different ecosystems, not only in Rust one. So it may produce some difficulties, but I suppose they won't be critical

**Comment by xpe** [comment]
@Undin Thanks for creating that issue (with screenshot and all!). I hope I can take a look at helping on it (but I don't have a timeframe in mind).
